### PR TITLE
API Add ValidationInterface to Form

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/11457

### CODE REVIEW

File: tests/php/Forms/FormTest.php [hunk 1]
```diff
@@ -1364,7 +1364,7 @@ public function testValidateFormFields(array $values, bool $isValid)
         }
         $form = new Form(null, 'testForm', $fieldList, new FieldList([/* no actions */]));
 
-        $result = $form->validationResult();
+        $result = $form->validate();
```
Comment (emteknetnz) [inline tests/php/Forms/FormTest.php:1367]: A couple of unit tests still called validationResult() which were missed in the CMS 5 PRs

Comment (GuySartorelli) [reply tests/php/Forms/FormTest.php:1367]: Should these be resolved in a new CMS 5 PR then?

**Review by GuySartorelli — COMMENTED** [review]
In https://github.com/silverstripe/silverstripe-framework/pull/11515#pullrequestreview-2510041881 I mentioned there seem to be calls to `validationResult()` from other modules. Can you please check if that's still the case before I merge this?

File: tests/php/Forms/FormTest.php (hunk 1 above)
Comment (emteknetnz) [reply tests/php/Forms/FormTest.php:1367]: Probably should, will do

Comment (emteknetnz) [reply tests/php/Forms/FormTest.php:1367]: Actually upon further inspection no they shouldn't

https://github.com/silverstripe/silverstripe-framework/commit/283d69a58188e2c23aac02768f2bbf4b56eb1bc6#diff-b26e65943a8d45b13830a9dc8b0a8471b23747fc12bf278afa7d484efcb33e7eR1367

Was introduced by a PR targetting 6.0, code isn't present in 5

https://github.com/silverstripe/silverstripe-framework/blame/5/tests/php/Forms/OptionsetFieldTest.php#L64

Code was changed to validate() in CMS 5 PR, looks was changed validationResult() in 6.0 as part of resolving a merge-up

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**